xc7/primitives/ff/ff.pb_type.xml

```diff
@@ -45,13 +45,13 @@
           <clock  name="C"  num_pins="1"/>
           <input  name="S"  num_pins="1"/>
           <output name="Q"  num_pins="1"/>
```

Comment:
What is preventing us either;
 (a) Renaming the port names in the XML to match the SDF file? IE
```xml
        <T_setup    value="{setup_CLK_DIN}"   clock="CLK" port="DIN" />
	<T_setup    value="{setup_CLK_CE}"    clock="CLK" port="CE"  />
	<T_setup    value="{recovery_CLK_SR}" clock="CLK" port="SR"  />
	<T_hold     value="{hold_CLK_DIN}"    clock="CLK" port="DIN" />
	<T_hold     value="{hold_CLK_CE}"     clock="CLK" port="CE"  />
	<T_hold     value="{removal_CLK_SR}"  clock="CLK" port="SR"  />
	<T_clock_to_Q max="{iopath_CLK_Q}"    clock="CLK" port="Q"   />
```
 (b) Rewriting the SDF file to match these names?

Reply:
the problem is that the bel timing fuzzer was bugged and did not emit those timings correctly. This merge https://github.com/SymbiFlow/prjxray/commit/8cb7366fa2c361fd5c8eec0aa0cb976808a39752 fixes that. Current DB was generated before the fix has been applied so it's missing e.g `setup_CLK_DIN`.

Reply:
@kgugala I don't think that answers my question? I was assuming you had the values like `setup_CLK_DIN` available to substitute?

Reply:
not in the current database. Because of the bug some timingchecks were emitted as e.g. `setup_CLK_CLK` or not emitted at all. This is now fixed, but not present in DB yet. Locally I run it with sdfs generated with the latest fuzzer and I can substitute all the timings tags in this PR. 

Answering your question: we need to update sdfs in the DB to have those values.

Reply:
My question is **not** about the values.

My question is about the names. Is there any reason we can't use the same names in both the XML and SDF files?

Reply:
The names have to be different in at least one case, the SR timing is shared between the FDRE and FDSE, and the pin names must match the blackbox.  In theory we could handle this at the techmap, but I don't think that is a good idea.

Reply:
ah. 
No, I don't think there is a reason preventing us from using the same names. I think rewriting xmls will be easier. The names in sdfs are in fact the Vivado names. Unfortunately those are not always consistent - e.g. the `D` input of a dff is named `DIN` in timingchecks and `D` in iopaths (in Latch configuration). BEL timing fuzzer could be updated to make the pins consistent.

I'd do the xml pin renaming in a separate PR so it is reviewable

Reply:
> I'd do the xml pin renaming in a separate PR so it is reviewable

Renaming the XML is not good enough, because the blackbox pin names must match the eblif.  I don't think I recommend changing the XML at all.

Reply:
Where do the blackbox pin names come from?

Reply:
The eblif and the XML must match.  The eblif is generated from Yosys (generally speaking).